x/ibc/04-channel/keeper/handshake.go

```diff
@@ -236,16 +250,13 @@ func (k Keeper) ChanOpenConfirm(
 		)
 	}
 
-	// TODO: blocked by #5542
-	// capkey, found := k.GetChannelCapability(ctx, portID, channelID)
-	// if !found {
-	// 	return sdkerrors.Wrap(types.ErrChannelCapabilityNotFound, channelID)
-	// }
+	if _, ok := k.GetChannelCapability(ctx, portID, channelID); !ok {
+		return sdkerrors.Wrap(types.ErrChannelCapabilityNotFound, "channel capability not found")
+	}
 
-	// key := sdk.NewKVStoreKey(capkey)
-	// if !k.portKeeper.Authenticate(key, portID) {
-	// 	return sdkerrors.Wrap(port.ErrInvalidPort, portID)
-	// }
+	if !k.portKeeper.Authenticate(ctx, portCap, portID) {
```

Comment:
This can be removed, the channel capability is used for `ChanOpenAck` and `ChanOpenConfirm`